File: frontend/src/types/country.ts
import type { UUID } from "./general";

export type CountryComponentId = `country_${UUID}`;
export type ElectionMode = "presidential" | "parliamentary";

export interface Country {
    id: UUID;
    componentId: CountryComponentId;
    label: string;
    electionMode: ElectionMode;
    descr: string;
    totalSeats: number;
}
export interface CountryResult {
    id: UUID;
    componentId: CountryComponentId;
    label: string;
}

File: backend/domain/country_schemas/country.py
from typing import Optional
from uuid import UUID

from pydantic import BaseModel

from domain.country_schemas.country_metrics import CountryMetrics
from domain.country_schemas.types import ElectionMode


class Country(BaseModel):
    id: UUID
    componentId: str
    label: str
    electionMode: ElectionMode
    descr: str
    totalSeats: int
    countryMetrics: Optional[CountryMetrics]= None
